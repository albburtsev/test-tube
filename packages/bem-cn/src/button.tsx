import * as React from 'react'
import { block } from 'bem-cn'

const b = block('button')

export interface ButtonProps {
	disabled?: boolean
}

export class Button extends React.Component<ButtonProps, {}> {
	render() {
		const { disabled, children } = this.props

		return (
			<button className={b({ disabled }).toString()} disabled={disabled}>
				{children}
			</button>
		)
	}
}
