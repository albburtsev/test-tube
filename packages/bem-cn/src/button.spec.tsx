import * as React from 'react'
import { Button } from './button'
import * as renderer from 'react-test-renderer'

test('Button component', () => {
	const tree = renderer.create(<Button disabled />)

	expect(tree.toJSON()).toMatchSnapshot()
})
