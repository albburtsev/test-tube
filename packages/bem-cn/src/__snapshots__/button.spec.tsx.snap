// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component 1`] = `
<button
  className="button button_disabled"
  disabled={true}
/>
`;
